tests/SemverTest.php

```diff
@@ -0,0 +1,309 @@
+     * @covers ::neq
+     * @dataProvider \Composer\Semver\Test\ComparatorTest::notEqualToProvider
+     * @uses \Composer\Semver\Comparator::notEqualTo
+     *
+     * @param string $version1
+     * @param string $version2
+     * @param bool $expected
+     */
+    public function testNotEqualTo($version1, $version2, $expected)
+    {
+        $this->assertEquals($expected, Semver::neq($version1, $version2));
+    }
+
+    /**
+     * @covers ::satisfies
+     * @dataProvider satisfiesProvider
+     *
+     * @param string $version
+     * @param string $constraint
+     * @param bool $expected
+     */
+    public function testSatisfies($version, $constraint, $expected)
+    {
+        $this->assertEquals($expected, Semver::satisfies($version, $constraint));
+    }
+
+    /**
+     * @return array
+     */
+    public function satisfiesProvider()
+    {
+        $positive = array_map(function ($array) {
+            $array[] = true;
+            return $array;
+        }, $this->satisfiesProviderPositive());
+
+        $negative = array_map(function ($array) {
+            $array[] = false;
+            return $array;
+        }, $this->satisfiesProviderNegative());
+
+        return array_merge($positive, $negative);
+    }
+
+    /**
+     * @return array
+     */
+    public function satisfiesProviderPositive()
+    {
+        return array(
+            array('1.2.3', '1.0.0 - 2.0.0'),
+//            array('1.2.3', '^1.2.3+build'),
+//            array('1.3.0', '^1.2.3+build'),
+//            array('1.2.3', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3-pre+asdf - 2.4.3pre+asdf'),
+//            array('1.2.3', '1.2.3pre+asdf - 2.4.3pre+asdf'),
+//            array('1.2.3-pre.2', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('2.4.3-alpha', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3+asdf - 2.4.3+asdf'),
+            array('1.0.0', '1.0.0'),
+//            array('0.2.4', '>=*'),
+//            array('', '1.0.0'),
+            array('1.2.3', '*'),
+            array('v1.2.3', '*'),
+            array('1.0.0', '>=1.0.0'),
+            array('1.0.1', '>=1.0.0'),
+            array('1.1.0', '>=1.0.0'),
+            array('1.0.1', '>1.0.0'),
+            array('1.1.0', '>1.0.0'),
+            array('2.0.0', '<=2.0.0'),
+            array('1.9999.9999', '<=2.0.0'),
+            array('0.2.9', '<=2.0.0'),
+            array('1.9999.9999', '<2.0.0'),
+            array('0.2.9', '<2.0.0'),
+            array('1.0.0', '>= 1.0.0'),
+            array('1.0.1', '>=  1.0.0'),
+            array('1.1.0', '>=   1.0.0'),
+            array('1.0.1', '> 1.0.0'),
+            array('1.1.0', '>  1.0.0'),
+            array('2.0.0', '<=   2.0.0'),
+            array('1.9999.9999', '<= 2.0.0'),
+            array('0.2.9', '<=  2.0.0'),
+            array('1.9999.9999', '<    2.0.0'),
+//            array('0.2.9', '<\t2.0.0'),
```

Comment:
This shold be in double quotes to have the tab expanded, but not sure if we support it or not. I guess it's harmless to support though